service/history/replication/dlq_handler.go

```diff
@@ -334,17 +339,14 @@ func (r *dlqHandlerImpl) getOrCreateTaskExecutor(clusterName string) (TaskExecut
 	if err != nil {
 		return nil, err
 	}
-	taskExecutor := NewTaskExecutor(
-		clusterName,
-		r.shard,
-		r.shard.GetNamespaceRegistry(),
-		r.resender,
-		engine,
-		r.deleteManager,
-		r.workflowCache,
-		r.shard.GetMetricsClient(),
-		r.shard.GetLogger(),
-	)
+	taskExecutor := r.taskExecutorProvider(TaskExecutorParams{
+		RemoteCluster:   clusterName,
```

Comment:
Maybe creating an issue for removing remote cluster name from task executor? So only one executor is needed per shard, instead of per shard per cluster. Then we can populate executor provider to fewer places.

Reply:
Add a TODO